wdl-lint/src/rules/command_shellcheck.rs

```diff
@@ -0,0 +1,361 @@
+
+        // Reset the visitor upon document entry
+        *self = Default::default();
+    }
+
+    fn command_section(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        section: &CommandSection,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        if !program_exists(SHELLCHECK_BIN) {
+            return;
+        }
+
+        // Collect declarations so we can ignore placeholder variables
+        let parent_task = section.parent().into_task().expect("parent is a task");
+        let mut decls = gather_task_declarations(&parent_task);
+
+        // Replace all placeholders in the command with
+        // dummy bash variables
+        let mut sanitized_command = String::new();
+        if let Some(cmd_parts) = section.strip_whitespace() {
+            cmd_parts.iter().for_each(|part| match part {
+                StrippedCommandPart::Text(text) => {
+                    sanitized_command.push_str(text);
+                }
+                StrippedCommandPart::Placeholder(placeholder) => {
+                    let bash_var = to_bash_var(placeholder);
+                    // we need to save the var so we can suppress later
+                    decls.insert(bash_var.clone());
+                    let mut expansion = String::from("\"$");
+                    expansion.push_str(&bash_var);
+                    expansion.push('"');
+                    sanitized_command.push_str(&expansion);
+                }
+            });
+        } else {
+            // This is the case where the command section contains
+            // mixed indentation. We silently return and allow
+            // the mixed indentation lint to report this.
+            return;
+        }
+
+        // get leading whitespace so we can add it to each span
+        let leading_whitespace = {
+            if let Some(first_part) = section.parts().next() {
+                match first_part {
+                    CommandPart::Text(text) => {
+                        let text_str = text
+                            .as_str()
+                            .strip_prefix("\n")
+                            .unwrap_or_else(|| text.as_str());
+                        count_leading_whitespace(text_str)
+                    }
+                    CommandPart::Placeholder(pholder) => {
+                        let pholder_str = &pholder.syntax().text().to_string();
+                        count_leading_whitespace(
+                            pholder_str.strip_prefix("\n").unwrap_or(pholder_str),
+                        )
+                    }
+                }
+            } else {
+                0
+            }
+        };
+
+        // Map each actual line of the command to its corresponding
+        // `CommandPart` and start position.
+        let mut line_map = HashMap::new();
+        let mut on_same_line = false;
+        let mut line_num = 0usize;
+        for part in section.parts() {
```

Comment:
FYI this algorithm is broken in all but the simplest case. Placeholder placement (and spacing) relative to a shellcheck lint causes the diagnostic to be misplaced. Working on updating it.